internal/message/types.go

```diff
@@ -7,120 +7,42 @@ package message
 
 import (
 	"github.com/telepresenceio/watchable"
-	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
 	gwapiv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
 	gwapiv1b1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
+	"github.com/envoyproxy/gateway/internal/gatewayapi"
 	"github.com/envoyproxy/gateway/internal/ir"
 	xdstypes "github.com/envoyproxy/gateway/internal/xds/types"
 )
 
 // ProviderResources message
 type ProviderResources struct {
-	GatewayClasses watchable.Map[string, *gwapiv1b1.GatewayClass]
-	Gateways       watchable.Map[types.NamespacedName, *gwapiv1b1.Gateway]
-	HTTPRoutes     watchable.Map[types.NamespacedName, *gwapiv1b1.HTTPRoute]
-	TLSRoutes      watchable.Map[types.NamespacedName, *gwapiv1a2.TLSRoute]
-	Namespaces     watchable.Map[string, *corev1.Namespace]
-	Services       watchable.Map[types.NamespacedName, *corev1.Service]
-	Secrets        watchable.Map[types.NamespacedName, *corev1.Secret]
-
-	ReferenceGrants watchable.Map[types.NamespacedName, *gwapiv1a2.ReferenceGrant]
+	GatewayAPIResources watchable.Map[string, *gatewayapi.Resources]
```

Comment:
It'd be helpful to add a godoc comment indicating the key to this map is a GatewayClass name.